src/memtable.rs

```diff
@@ -100,17 +150,34 @@ pub struct MemTable {
 
     /// Shared statistics.
     global_stats: Arc<GlobalStats>,
+
+    _phantom: PhantomData<A>,
 }
 
-impl MemTable {
-    pub fn new(region_id: u64, global_stats: Arc<GlobalStats>) -> MemTable {
+impl MemTable<VacantAllocator> {
+    #[allow(dead_code)]
+    fn new(region_id: u64, global_stats: Arc<GlobalStats>) -> MemTable<VacantAllocator> {
+        Self::with_allocator(region_id, global_stats, &new_vacant_allocator())
+    }
+}
+
+impl<A: AllocatorTrait> MemTable<A> {
+    fn with_allocator(
+        region_id: u64,
+        global_stats: Arc<GlobalStats>,
+        _allocator: &A,
+    ) -> MemTable<A> {
         MemTable {
             region_id,
+            #[cfg(feature = "swap")]
```

Comment:
It's really necessary to use feature?

Reply:
`with_capacity_in` is a nightly interface. We already took some efforts to make raft-engine a stable crate, so `swap` feature is needed to gate these nightly-only stuff.